src/commands/timeout.ts

```diff
@@ -0,0 +1,160 @@
+                    option.setName('user').setDescription('User to remove the timeout from').setRequired(true),
+                )
+                .addStringOption((option) =>
+                    option.setName('reason').setDescription('Reason for removing the timeout').setRequired(false),
+                ),
+        ),
+    run: async (interaction: ChatInputCommandInteraction) => {
+        // Typeguard in order to ensure having access to ChatInputCommand interaction options.
+        if (!interaction.isChatInputCommand()) return;
+
+        if (interaction.inGuild() === false) {
+            return interaction.reply({ content: "❌ I can't execute this command inside DMs!", ephemeral: true });
+        }
+
+        if (!interaction.guild.members.me.permissions.has(PermissionFlagsBits.ModerateMembers)) {
+            return interaction.reply({
+                content: '❌ Sorry, I need the `Moderate Members` in order to execute this command.',
+                ephemeral: true,
+            });
+        }
+        if (typeof interaction.member.permissions === 'string')
+            return interaction.reply({ content: '❌ Unknown permissions. Please try again later.', ephemeral: true });
+        else if (!interaction.member.permissions.has(PermissionFlagsBits.ModerateMembers)) {
+            return interaction.reply({
+                content: '⛔ You need the `Moderate Members` permission in order to use this command!',
+                ephemeral: true,
+            });
+        }
+
+        const user = interaction.options.getUser('user', true);
+        const member = await interaction.guild.members.fetch(user.id);
+
+        if (!member) {
+            return interaction.reply({
+                content: '❌ You need to specify a valid user to manage their timeout!',
+                ephemeral: true,
+            });
+        }
+
+        if (interaction.options.getSubcommand() === 'add') {
+            if (member.id === interaction.user.id) {
+                return interaction.reply({ content: "Aww, please don't time out yourself! 💖", ephemeral: true });
+            }
+
+            if (member.moderatable === false) {
+                return interaction.reply({
+                    content:
+                        '❌ I cannot time out this user! \n**Please make sure that my highest role is above theirs.**',
+                    ephemeral: true,
+                });
+            }
+
+            const duration = interaction.options.getString('duration', true);
+            const msDuration = ms(duration);
+
+            if (msDuration == null) {
+                return interaction.reply({
+                    content: '❌ The format of the timeout duration is invalid!',
+                    ephemeral: true,
+                });
+            }
+
+            if (msDuration <= 0) {
+                return interaction.reply({
+                    content: '❌ The duration of the timeout must be greater than 0!',
```

Comment:
I'd rather see this string changed to the one below.
```suggestion
                    content: '❌ Please specify a valid timeout duration.',
```

Reply:
Done, however, wouldn't you like an exclamation mark at the end? As it is as error message.

Reply:
Not all error messages have exclamation marks at the end, whether or not to include one is decided by how it sounds with or without one when reading the error message out loud. I don't see a need for a question mark here, personally.

Reply:
Okay, I'll go with the . instead of an !